xbmc/URL.cpp

```diff
@@ -333,15 +333,15 @@ void CURL::SetFileName(const std::string& strFileName)
 {
   m_strFileName = strFileName;
 
-  int slash = m_strFileName.find_last_of(GetDirectorySeparator());
-  int period = m_strFileName.find_last_of('.');
-  if(period != -1 && (slash == -1 || period > slash))
+  size_t slash = m_strFileName.find_last_of(GetDirectorySeparator());
+  size_t period = m_strFileName.find_last_of('.');
+  if(period != string::npos && (slash == string::npos || period > slash))
     m_strFileType = m_strFileName.substr(period+1);
   else
     m_strFileType = "";
 
   slash = m_strFileName.find_first_of(GetDirectorySeparator());
-  if(slash == std::string::npos)
+  if(slash == string::npos)
```

Comment:
My take on using, in the case of std here, is that, it's more or less OK, though I agree it's not the best solution, my preference would be to get the scope to a minimum (i.e. in the method) and let the resolution do its work but that can easily get out of hand too for common stuff, like string. On the other hand, IMO, explicit namespacing also has its ups and downs. I will change it in a follow up commit to keep up with the project standard.